contracts/IdleCDOTrancheRewards.sol

```diff
@@ -140,11 +146,17 @@ contract IdleCDOTrancheRewards is Initializable, PausableUpgradeable, OwnableUpg
       }
       // Set the user index equal to the global one, which means 0 rewards
       usersIndexes[msg.sender][reward] = adjustedRewardIndex(reward);
+      _addRewardNotTransferred(reward, usersStakes[msg.sender]);
       // transfer the reward to the user
       IERC20Detailed(reward).safeTransfer(msg.sender, amount);
     }
   }
 
+  function _addRewardNotTransferred(address reward, uint256 userStake) internal {
+    uint256 _unClaimableRewards = _unClaimableRewards(reward);
+    indexToRewardNotTransferred[rewardsIndexes[reward]] += _unClaimableRewards * userStake / totalStaked;
```

Comment:
> I'm not sure if this can always be correct without any reference to the reward address, I think that potentially
there could be a clash between reward indexes
Nice. As you say, this should refer to the reward address.
 
> given that you don't really need the reward index, wdyt?
If we use  `mapping (address => mapping(uint256 => uint256)) public indexToRewardNotTransferred;`,
the stored rewards will be reset the next time deposiReward is called, so users still in the pool won't be able to earn them.  I think `  mapping (address => uint256) public rewardToRewardsNoTransferred` is better as you suggested.